polars/polars-core/src/testing.rs

```diff
@@ -79,6 +79,9 @@ impl DataFrame {
                 return false;
             }
         }
+        if self.schema() != other.schema() {
+            return false;
```

Comment:
Will now also check if the schemas match.

Reply:
A schema needs to be materialized which is a needless allocation. We can better iterate the colums and check if all `dtype == dtype`. (Name is already checked in `series_equal`.